src/mixins/raster-layer-cross-section-terrain-mixin.js

```diff
@@ -0,0 +1,325 @@
+    if (typeof setter === "function") {
+      state = setter(state)
+    } else {
+      state = setter
+    }
+    return _layer
+  }
+
+  _layer.getState = function() {
+    return state
+  }
+
+  // NOTE: as of 11/14/22 the "getTransforms" method that is found in most of the other raster layer mixin classes
+  // only seems to be called from immerse via the buildRasterExportSql() method in raster-sql.ts
+  //
+  // It is not currently believed that Mesh2d marks need to support export, so the "getTransforms" method is not
+  // going to be included
+
+  const prop_descriptors = new Map()
+
+  prop_descriptors.set("x", new PositionChannelDescriptor("x"))
+  prop_descriptors.set("y", new PositionChannelDescriptor("y"))
+
+  prop_descriptors.set(
+    "longitude",
+    new GeographicChannelDescriptor("longitude", "x")
+  )
+
+  prop_descriptors.set(
+    "latitude",
+    new GeographicChannelDescriptor("latitude", "y")
+  )
+
+  prop_descriptors.set(
+    "strokeColor",
+    new StringPropDescriptor(
+      "strokeColor",
+      null,
+      PropLocation.kMarkDefOnly,
+      null,
+      false
+    )
+  )
+  prop_descriptors.set(
+    "strokeWidth",
+    new NumericPropDescriptor(
+      "strokeWidth",
+      null,
+      PropLocation.kMarkDefOnly,
+      null,
+      false,
+      is_zero_to_one
+    )
+  )
+  prop_descriptors.set(
+    "fillBelowLine",
+    new NumericPropDescriptor(
+      "fillBelowLine",
+      null,
+      PropLocation.kMarkDefOnly,
+      null,
+      false,
+      is_zero_to_one
+    )
+  )
+
+  _layer.__genVega = function({
+    chart,
+    table,
+    filter,
+    lastFilteredSize,
+    globalFilter,
+    // eslint-disable-next-line no-unused-vars
+    pixelRatio,
+    layerName
+  }) {
+    const raster_layer_context = new RasterLayerContext(
+      chart,
+      table,
+      CrossSectionTerrainConfigDefinitionObject.key,
+      this,
+      layerName,
+      lastFilteredSize
+    )
+
+    _vega_property_output_state = materializePropDescriptors(
+      raster_layer_context,
+      prop_descriptors,
+      state
+    )
+    const {
+      sql_parser_transforms,
+      vega_data_formats,
+      vega_transforms,
+      vega_scales,
+      mark_properties
+    } = _vega_property_output_state.flatten()
+
+    assert(vega_data_formats.length <= 1)
+
+    if (typeof filter === "string" && filter.length) {
+      sql_parser_transforms.push({
+        type: "filter",
+        expr: filter
+      })
+    }
+
+    const post_filter_transform = create_post_filter_transform(
+      state.postFilters
+    )
+
+    if (post_filter_transform) {
+      sql_parser_transforms.push(post_filter_transform)
+    }
+
+    if (typeof globalFilter === "string" && globalFilter.length) {
+      sql_parser_transforms.push({
+        type: "filter",
+        expr: globalFilter
+      })
+    }
+
+    const data = [
+      {
+        name: layerName,
+        ...(vega_data_formats.length ? vega_data_formats[0] : {}),
```

Comment:
minor, but can make this a little simpler: `vega_data_formats.?[0] ?? {}`

Reply:
Good call!